internal/pkg/util/fs/fuse/fuse_mount_linux.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package fuse
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/sylabs/singularity/v4/internal/pkg/util/bin"
+	"github.com/sylabs/singularity/v4/pkg/image"
+	"github.com/sylabs/singularity/v4/pkg/sylog"
+)
+
+type ImageMount struct {
+	// Type represents what type of image this mount involves (from among the
+	// values in pkg/image)
+	Type int
+
+	// Readonly represents whether this is a Readonly overlay
+	Readonly bool
+
+	// SourcePath is the path of the image, stripped of any colon-prefixed
+	// options (like ":ro")
+	SourcePath string
+
+	// EnclosingDir is the location of a secure parent-directory in
+	// which to create the actual mountpoint directory
+	EnclosingDir string
```

Comment:
This doesn't necessarily make sense now that this is a much more generic fuse mounting routine. The need for a secure parent directory is not universal. It should probably be the callers responsibility.